routes/degree_days.py

```diff
@@ -0,0 +1,375 @@
+    "projected": {"min": 2006, "max": 2100},
+}
+
+
+def get_dd_wcps_request_str(x, y, cov_id, models, years, tempstat, encoding="json"):
+    """Generates a WCPS query specific to the
+    coverages used for the degree days min-mean-max.
+
+    Args:
+        x (float or str): x-coordinate for point query, or string
+            composed as "x1:x2" for bbox query, where x1 and x2 are
+            lower and upper bounds of bbox
+        y (float or str): y-coordinate for point query, or string
+            composed as "y1:y2" for bbox query, where y1 and y2 are
+            lower and upper bounds of bbox
+        cov_id (str): Rasdaman coverage ID
+        models (str): Comma-separated numbers of requested models
+        years (str): Colon-separated full date-time i.e.
+            "\"2006-01-01T00:00:00.000Z\":\"2100-01-01T00:00:00.000Z\""
+        tempstat(int): Integer between 0-2 where:
+            - 0 = ddmax
+            - 1 = ddmean
+            - 2 = ddmin
+        encoding (str): currently supports either "json" or "netcdf"
+            for point or bbox queries, respectively
+
+    Returns:
+        WCPS query to be included in generate_wcs_url()
+    """
+    if tempstat == 0:
+        operation = "max"
+    elif tempstat == 2:
+        operation = "min"
+    else:
+        operation = "+"
+
+    if tempstat == 0 or tempstat == 2:
+        wcps_request_str = quote(
+            (
+                f"ProcessCoverages&query=for $c in ({cov_id}) "
+                f"let $a := {operation}(condense {operation} over $m model({models}) "
+                f"using $c[model($m),year({years}),X({x}),Y({y})] ) "
+                f'return encode( $a, "application/{encoding}")'
+            )
+        )
+        return wcps_request_str
+    else:
+        # Generates the mean across models
+
+        # For projected, 2 models
+        num_results = 2
+
+        # For historical, only a single model
+        if models == "0:0":
+            num_results = 1
+
+        wcps_request_str = quote(
+            (
+                f"ProcessCoverages&query=for $c in ({cov_id}) "
+                f"let $a := avg(condense {operation} over $m model({models}) "
+                f"using $c[model($m),year({years}),X({x}),Y({y})] / {num_results} ) "
+                f'return encode( $a, "application/{encoding}")'
+            )
+        )
+        return wcps_request_str
+
+
+@routes.route("/mmm/degree_days/")
+@routes.route("/mmm/degree_days/abstract/")
+@routes.route("/mmm/degree_days/heating/")
+@routes.route("/mmm/degree_days/below_zero/")
+@routes.route("/mmm/degree_days/thawing_index/")
+@routes.route("/mmm/degree_days/freezing_index/")
+def heating_degree_days_about():
+    return render_template("/mmm/degree_days.html")
+
+
+def package_dd_point_data(point_data, var_ep, horp):
+    """Add data for JSON response
+
+    Args:
+        point_data (list): nested list containing JSON
+            results of WCPS query
+        var_ep (str): variable name
```

Comment:
minor nitpick, add arg for `horp` to docstring

Reply:
Fixed in 352c2027d213a687a5433fad95548ea0e6ba52fd.